jmespath-core/src/main/java/io/burt/jmespath/function/NotNullFunction.java

```diff
@@ -0,0 +1,21 @@
+package io.burt.jmespath.function;
+
+import java.util.List;
+
+import io.burt.jmespath.Adapter;
+
+public class NotNullFunction extends JmesPathFunction {
+  public NotNullFunction() {
+    super(1, Integer.MAX_VALUE);
+  }
+
+  @Override
+  protected <T> T internalCall(Adapter<T> adapter, List<ExpressionOrValue<T>> arguments) {
+    for (ExpressionOrValue<T> x : arguments) {
+      if (!adapter.isNull(x.value())) {
+        return x.value();
+      }
```

Comment:
I'm planning on making a type checker for the functions that is configured via annotations, but I don't want to get bogged down with that right now. I'd like it to work something like this, but I'm not yet sure how much is possible with Java annotations:

```java
@JmesPathFunction(arguments = {StringArgument, NumberArgument})
public class FooFunction extends JmesPathFunction { }

@JmesPathFunction(name = "barf_barf", arguments = {AnyArgument})
public class BurfFunction extends JmesPathFunction { }


@JmesPathFunction(arguments = {ArrayArgument(AnyArgument)})
public class FnurbFunction extends JmesPathFunction { }
```

I think describing things like "array of strings" or even "array of strings or numbers" will be problematic unfortunately.